src/backend/catalog/pg_depend.c

```diff
@@ -787,3 +797,119 @@ get_index_constraint(Oid indexId)
 
 	return constraintId;
 }
+
+/*
+ * depLockAndCheckObject
+ *
+ * Lock the object that we are about to record a dependency on.
+ * After it's locked, verify that it hasn't been dropped while we
+ * weren't looking.  If the object has been dropped, this function
+ * does not return!
+ */
+static void
+depLockAndCheckObject(const ObjectAddress *referenced)
+{
+	int			cacheId = -1;
+	char	   *objName;
```

Comment:
may be
```suggestion
	enum SysCacheIdentifier cacheId;
	const char *objName;
```
and remove assert below?